### PR TITLE
feat(sql): coalesce function

### PR DESCRIPTION
**Pull request by ideoma**

### CODE REVIEW

File: core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+        }
+
+        @Override
+        public ObjList<Function> getArgs() {
+            return args;
+        }
+
+        @Override
+        public double getDouble(Record rec) {
+            for (int i = 0, size = args.size(); i < size; i++) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java:103]: `args.size()` should not be evaluated for each record

File: core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+                double value = args.getQuick(i).getDouble(rec);
+                if (!Double.isNaN(value)) {
+                    return value;
+                }
+            }
+            return Double.NaN;
+        }
+    }
+
+    private static class FloatCoalesceFunction extends FloatFunction implements MultiArgFunction {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java:113]: we should also have implementations for two arguments, e.g. `coalesce(a,b)`. In this case we do not need setup a loop thus making impl faster

File: core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+            return args;
+        }
+
+
+        @Override
+        public CharSequence getStr(Record rec) {
+            for (int i = 0, size = args.size(); i < size; i++) {
+                Function arg = args.getQuick(i);
+                CharSequence value;
+                if (arg.getType() == ColumnType.SYMBOL) {
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java:314]: a suggestion: perhaps we can detect if all types are homogeneous or not and have more implementation for:
- all strings
- all symbols
- mixed

I think it is less likely different types are mixed. I'm open to discussion.

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java (hunk 1 above)
Comment (ideoma) [reply core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java]: done

File: core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java (hunk 2 above)
Comment (ideoma) [reply core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java:202]: done

File: core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java (hunk 3 above)
Comment (ideoma) [reply core/src/main/java/io/questdb/griffin/engine/functions/conditional/CoalesceFunctionFactory.java]: done

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**